### DISCUSSION TITLE
Using Veramo with Ganache

### DISCUSSION THREAD

**Discussion by AbderrazagB** [decentralized-identity/veramo#1363] [Q&A]
Hey, Im new to vermao and im trying to find a way to use in a local node (ganache) but i cant seem to connect to it.
I will be very grateful if someone could help me!

**Comment by iyedM** [comment]
Same problem

**Comment by pauldesmondparker** [comment]
This might be a bit late for you guys, but I'll answer.  Ganache has been sunset by ConsenSys, so for a local node you'd be looking to use `hardhat node` or `anvil` (forge).

## Deploy
Once you have one of those local networks up, you'd first want to deploy your registry flavor, for example [`ethr-did-registry`](https://github.com/uport-project/ethr-did-registry).

At the bottom of the README there are instructions to deploy the registry to a blockchain. If you need more detail look at this [issue](https://github.com/uport-project/ethr-did-registry/issues/63).

## Configure
You'll need to look in the docs for the context of where this should go, but this is a rough go at it.
```ts
          new DIDManager({
            store: new DIDStore(dbConnection),
            defaultProvider: 'anvil',
            providers: {
              'did:ethr:anvil': new EthrDIDProvider({
                defaultKms: 'local',
                networks: [
                  {
                    name: 'anvil'  // I would have named this `local` but for the confusion with the local kms.
                    provider: new JsonRpcProvider( 
                      127.0.0.1:8545,  // Anvil  local node
                      31337,  // ChainId, not tested by me
                    ),
                    registry: '0x03d5003bf0e79c5f5223588f347eba39afbc3818',  // current version CREATE2 address
                  },
                ],
              }),
            },
          }),
```
You'd also need to setup your `DIDResolverPlugin`.

I prefer to use `JsonRpcProvider` above because I gain extra control over the additional settings that it takes. Specifically, `staticNetwork` which stops ethers making a ridiculous number of `eth_chainid` calls against a rate and/or usage limited rpc endpoint. Not needed for a local node though.

Your dids will look like this `did:ethr:anvil:0x03fea...a975`

**Reply by mirceanis** [reply]
Great stuff!

Perhaps we should replace the tests that depend on ganache with something like this.
Is it possible to spin-up anvil or hardhat node in a plain jest test?
